docs/sdk/src/reference_docs/omni_node.rs

```diff
@@ -0,0 +1,170 @@
+//! node, and a runtime.
+//!
+//! Moreover, the code and API of each of these nodes was historically very advanced, and tailored
+//! towards those who wish to customize many of the node components at depth.
+//!
+//! <details>
+//! <summary>
+//! The notorious <pre>service.rs</pre> is a good example of this:
+//! </summary>
+// #![doc = docify::embed!("../../substrate/bin/node/src/cli/service.rs", "new_full_base")]
+//!
+//! </details>
+//!
+//! A [trend](https://github.com/paritytech/polkadot-sdk/issues/62) has already been undergoing in
+//! order to de-couple the node and the runtime for a long time. The north star of this effort is
+//! twofold :
+//!
+//! 1. develop what can be described as an "omni-node": A node that can run all runtimes.
+//! 2. provide a cleaner abstraction for creating a custom node.
+//!
+//! While a single omni-node running *all possible runtimes* is not feasible, the
+//! [`polkadot-omni-node`] is an attempt at creating the former, and the [`polkadot-omni-node-lib`]
+//! is the latter.
+//!
+//! > The OmniNodes are mainly focused on the development needs of Polkadot parachains, not
+//! > Substrate solo-chains. For the time being, solo-chains are not supported by the OmniNodes.
+//! > This might change in the future.
+//!
+//! ## `polkadot_omni_node`
+//!
+//! `polkadot_omni_node` is a white-labeled binary, released as a part of Polkadot SDK that is
+//! capable of meeting the needs of most Polkadot parachains. It can act as the collator of a
+//! parachain in production, and it can run the wasm blob of the parachain locally for testing and
+//! development.
+//!
+//! ## [`polkadot_omni_node_lib`]
+//!
+//! [`polkadot_omni_node_lib`] is the library version of the above, which can be used to create a
+//! fresh parachain node, with a some limited configuration options using a lean API.
+//!
+//! ## Old School Nodes
+//!
+//! The existing node architecture, as seen in the [`templates`], is still available for those who
+//! want to have full control over the node software.
+//!
+//! ## Summary
+//!
+//! We can summarize the choices for the node software of any given user of Polkadot-Sdk, wishing to
+//! deploy a parachain into 3 categories:
+//!
+//! 1. **Use the `polkadot-omni-node`**: This is the easiest way to get started, and is the most
+//!   likely to be the best choice for most users.
+//! 	* can run almost any runtime with `--dev-block-time`
+//! 	* can run a parachain runtime, in real mode.
+//! 2. **Use the `polkadot-omni-node-lib`**: This is the best choice for those who want to have
+//!    slightly more
+//!  control over the node software, such as embedding a custom chain-spec.
+//! 3. **Use the old school nodes**: This is the best choice for those who want to have full control
+//! 	over the node software, such as changing the consensus engine, altering the transaction pool,
+//!		and so on.
+//!
+//! ## User Journey
```

Comment:
nit: I have a mixed feelings here:
```suggestion
//! ## Omni-tools: user journey
```
My point here is that maybe we shall mention that this section is about related tooling?

Reply:
Will incorporate, thanks!